src/dtproject_tests/test_layer_hooks.py

```diff
@@ -0,0 +1,41 @@
+from dtproject.types import Hook, LayerHooks
+from . import get_project_path
+
+from dtproject import DTProject
+import unittest
+
+
+class TestLayerSelf(unittest.TestCase):
+
+    def test_layer_self_project_v1(self):
+        pd = get_project_path("basic_v1")
+        p = DTProject(pd)
+        # ---
+
+    def test_layer_self_project_v2(self):
+        pd = get_project_path("basic_v2")
+        p = DTProject(pd)
+        # ---
+        self.assertEqual(p.name, "basic_v2")
+        self.assertEqual(p.version, "0.0.0")
+
+    def test_layer_self_project_v3(self):
+        pd = get_project_path("basic_v3")
+        p = DTProject(pd)
+        # ---
+
+    def test_layer_self_project_v4(self):
+        pd = get_project_path("basic_v4")
+        p = DTProject(pd)
+        # ---
+    
+        
+    def test_layer_self_project_v4_custom(self):
+        pd = get_project_path("custom_v4")
+        p = DTProject(pd)
+        # ---
+        self.assertEqual(p.hooks,LayerHooks(version=2, hooks={'pre-build': [Hook(command='echo "pre-build hook"', required=True)], 'post-build': [Hook(command='echo "post-build hook"', required=True)],}))
```

Comment:
This is not true according to the content of the file `src/dtproject_tests/assets/projects/custom_v4/dtproject/hooks.yaml` above. In fact, it fails in the tests.

![image](https://github.com/duckietown/lib-dtproject/assets/22665009/4366a5ab-8c46-495d-9661-fd0dd2cf39d7)

Reply:
You're right I had modified it to have some further changes, fixing it now.